devito/propagator.py

```diff
@@ -299,6 +350,39 @@ def add_time_loop_stencil(self, stencil, before=False):
         else:
             self.time_loop_stencils_a.append(stencil)
 
+            # initialises block sizes as variables and adds auto tuning pragmas
+
+    def _at_init_block_vars(self):
+        """Initialises block sizes as variables in order to at them later"""
+
+        block_vars = []  # Blocking var names
+
+        for i in range(0, len(self.space_dims) - 1):  # generate block size vars. We want to ignore inner most dimension
+            block_vars.append(str(self.loop_counters[i]) + "block")
+
+        # main auto tuning pragma
+        at_main_pragma = "isat tuning name(spc_o_%s_tm_o_%s) scope(%s, %s) measure(%s, %s)" \
+                         % (self.spc_order, self.time_order, self.at_markers[0][0], self.at_markers[0][1],
+                            self.at_markers[1][0], self.at_markers[1][1])
+
+        for block_var in block_vars:  # appends vars that we want to tune to main at pragma
+            at_main_pragma += " variable(%s, range(%d, %d, 1))" \
+                              % (block_var, self.tune_b_size - self.tune_range, self.tune_b_size + self.tune_range)
+
+        # dependant will try all possible permutations of block sizes // prob what we want
+        # independant will find the first optimal varthen progress searching for the next one using the one found before
+        at_main_pragma += " search(dependent)"
```

Comment:
It is configurable by deleting/adding "in" form independent in code. I know that its not the best practise, however 99% of a time we want to run dependent, so I see no reason to actually make it configurable outside the method